File: go/groovy.go
package main

import (
	"fmt"
	"os"
	"time"
)

const dotDuration = 200 * time.Millisecond
const dashDuration = 3 * dotDuration
const symbolPauseDuration = dotDuration
const letterPauseDuration = 3 * dotDuration

var morseCode = map[rune]string{
	'A':  ".-",
	'B':  "-...",
	'C':  "-.-.",
	'D':  "-..",
	'E':  ".",
	'F':  "..-.",
	'G':  "--.",
	'H':  "....",
	'I':  "..",
	'J':  ".---",
	'K':  "-.-",
	'L':  ".-..",
	'M':  "--",
	'N':  "-.",
	'O':  "---",
	'P':  ".--.",
	'Q':  "--.-",
	'R':  ".-.",
	'S':  "...",
	'T':  "-",
	'U':  "..-",
	'V':  "...-",
	'W':  ".--",
	'X':  "-..-",
	'Y':  "-.--",
	'Z':  "--..",
	'0':  "-----",
	'1':  ".----",
	'2':  "..---",
	'3':  "...--",
	'4':  "....-",
	'5':  ".....",
	'6':  "-....",
	'7':  "--...",
	'8':  "---..",
	'9':  "----.",
	'.':  ".-.-.-",
	',':  "--..--",
	'?':  "..--..",
	'\'': ".----.",
	'!':  "-.-.--",
	'/':  "-..-.",
	'(':  "-.--.",
	')':  "-.--.-",
	'&':  ".-...",
	':':  "---...",
	';':  "-.-.-.",
	'=':  "-...-",
	'+':  ".-.-.",
	'-':  "-....-",
	'_':  "..--.-",
	'"':  ".-..-.",
	'$':  "...-..-",
	'@':  ".--.-.",
}

func main() {
	message := "SOS"
	for _, c := range message {
		if code, ok := morseCode[c]; ok {
			for _, s := range code {
				switch s {
				case '.':
					dot()
				case '-':
					dash()
				}
				time.Sleep(symbolPauseDuration)
			}
			time.Sleep(letterPauseDuration)
		}
	}
}

func dot() {
	fmt.Print(".")
	beep(dotDuration)
}

func dash() {
	fmt.Print("-")
	beep(dashDuration)
}

func beep(d time.Duration) {
	speaker, err := os.Open("/dev/audio")
	if err != nil {
		fmt.Println("Failed to open speaker:", err)
		return
	}
	defer speaker.Close()
	for t := 0; t < int(d)/10; t++ {
		speaker.Write([]byte{1, 1, 1, 1, 1, 1, 1, 1})
		time.Sleep(10 * time.Millisecond)
		speaker.Write([]byte{0, 0, 0, 0, 0, 0, 0, 0})
		time.Sleep(10 * time.Millisecond)
	}
}

File: Python/groovy.py
import winsound
import time

# モールス信号を辞書として定義
morse_code = {
    'A': '.-', 'B': '-...', 'C': '-.-.', 'D': '-..', 'E': '.',
    'F': '..-.', 'G': '--.', 'H': '....', 'I': '..', 'J': '.---',
    'K': '-.-', 'L': '.-..', 'M': '--', 'N': '-.', 'O': '---',
    'P': '.--.', 'Q': '--.-', 'R': '.-.', 'S': '...', 'T': '-',
    'U': '..-', 'V': '...-', 'W': '.--', 'X': '-..-', 'Y': '-.--',
    'Z': '--..', '1': '.----', '2': '..---', '3': '...--', '4': '....-',
    '5': '.....', '6': '-....', '7': '--...', '8': '---..', '9': '----.',
    '0': '-----', ' ': ' ', ',': '--..--', '.': '.-.-.-', '?': '..--..',
    '/': '-..-.', '-': '-....-', '(': '-.--.', ')': '-.--.-'
}

# メッセージを入力してもらい、モールス信号に変換する関数
def encode_message(message):
    encoded_message = ""
    for letter in message:
        if letter.upper() in morse_code:
            encoded_message += morse_code[letter.upper()] + " "
        else:
            encoded_message += " "
    return encoded_message

# ユーザーにメッセージを入力してもらい、モールス信号に変換する
message = input("Enter your message: ")
encoded_message = encode_message(message)

# モールス信号を音で再生する
for symbol in encoded_message:
    if symbol == " ":
        # スペースは0.5秒待つ
        time.sleep(0.5)
    elif symbol == ".":
        # ピッと鳴らす
        winsound.Beep(1000, 250)
    elif symbol == "-":
        # 長く鳴らす
        winsound.Beep(1000, 750)